pydicer/convert/pet_to_nifti_convert_new.py

```diff
@@ -0,0 +1,409 @@
+                           modality='CT',
+                           sitk_writer=None,
+                           rtstruct_file=None,
+                           labels_rtstruct=['GTVt'],
+                           patient_weight_from_ct=None,
+                           extension='.nii',
+                           dtype_image=np.float32,
+                           dtype_mask=np.uint8):
+    """Function to convert the dicom files contained in input_filepaths to one
+       NIFTI image.
+    Args:
+        input_filepaths (list): list of the dicom paths
+        output_folder (str): path to the output folder where to store the
+                             NIFTI file.
+        modality (str, optional): The modality of the DICOM, it is used to
+                                  obtain the correct physical values
+                                  (Hounsfield unit for the CT and SUV for the
+                                  PT). Defaults to 'CT'.
+        sitk_writer (sitk.WriteImage(), optional): The SimpleITK object used
+                                                   to write an array to the
+                                                   NIFTI format. Defaults to
+                                                   None.
+        rtstruct_file (str, optional): Path to the RTSTRUCT file associated
+                                       with the current image. Defaults to
+                                       None.
+        labels_rtstruct (list, optional): List of label to extract from the
+                                          RTSTRUCT. Defaults to ['GTVt'].
+        patient_weight_from_ct (float, optional): If the patient's weight is
+                                                  missing from the PT DICOM
+                                                  it can be provided through
+                                                  this argument. Defaults to
+                                                  None.
+        extension (str, optional): The extension in which to save the NIFTI.
+                                   Defaults to '.nii'.
+        dtype_image (numpy.dtype, optional): The dtype in which to save the
+                                             image. Defaults to np.float32.
+        dtype_mask (numpy.dtype, optional): The dtype in which to save the
+                                            segmentation. Defaults to np.uint8.
+    Raises:
+        MissingWeightException: Error to alert when the weight is missing from
+                                the PT, to compute the SUV.
+        RuntimeError: Error to alert when one or more slices are missing
+        ValueError: Raised when a modality or a unit (for the PT) is not
+                    handled.
+    Returns:
+        numpy.array: The numpy image, used to compute the bounding boxes
+    """
+    slices = [pdcm.read_file(dcm) for dcm in input_filepaths]
+    slices.sort(key=lambda x: float(x.ImagePositionPatient[2]))

```

Comment:
This will only work for images in standard orientation. Check out the preprocess function where Rob has provided some code which handles this for images in any orientation.